### PR TITLE
RAB-1096: Support Microsoft Azure remote server (front)

### PR DESCRIPTION
**Pull request by axelhms**
**Description (for Contributor and Core Developer)**

In this PR, I've added support for Microsoft Azure remote server.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by pierrejolly — APPROVED** [review]

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.test.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      validationErrors={[]}
+      onStorageChange={onStorageChange}
+    />
+  );
+
+  const checkButton = screen.getByText('pim_import_export.form.job_instance.connection_checker.label');
+  await act(async () => {
+    userEvent.click(checkButton);
+  });
+
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.test.tsx:232]: In order to check that there is no error message
```suggestion 
expect(screen.getByText('pim_import_export.form.job_instance.connection_checker.exception')).not.toBeInTheDocument();
```

May assert the success message also

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  flex-direction: column;
+  gap: 5px;
+`;
+
+const CheckStorageConnection = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 8.5px;
+  color: ${getColor('green', 100)};
+`;
```
[the remark this answers is not in the document]
Comment (StevenVAIDIE) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx:19]: :+1: it can be done in another PR if you want

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        })}
+        onChange={file_path => onStorageChange({...storage, file_path})}
+        errors={filterErrors(validationErrors, '[file_path]')}
+      />
+      <TextField
+        required={true}
+        value={storage.connection_string}
+        label={translate('pim_import_export.form.job_instance.storage_form.connection_string.label')}
+        placeholder={translate('pim_import_export.form.job_instance.storage_form.connection_string.placeholder')}
+        onChange={(connection_string: string) => onStorageChange({...storage, connection_string: connection_string})}
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx:51]: :hackerman: Strange that @ValentinMumble didn't see it :smile: 
```suggestion
        onChange={(connection_string: string) => onStorageChange({...storage, connection_string})}
```

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        errors={filterErrors(validationErrors, '[connection_string]')}
+      />
+      <TextField
+        required={true}
+        value={storage.container_name}
+        label={translate('pim_import_export.form.job_instance.storage_form.container_name.label')}
+        placeholder={translate('pim_import_export.form.job_instance.storage_form.container_name.placeholder')}
+        onChange={(container_name: string) => onStorageChange({...storage, container_name: container_name})}
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx:59]: Same here

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/model.ts [hunk 5]
```diff
@@ -75,11 +85,21 @@ const isAmazonS3Storage = (storage: Storage): storage is AmazonS3Storage => {
   );
 };
 
+const isMicrosoftAzureStorage = (storage: Storage): storage is MicrosoftAzureStorage => {
+  return (
+    'microsoft_azure' === storage.type &&
+    'connection_string' in storage &&
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/model.ts:91]: Maybe we can check that those values are string

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts [hunk 6]
```diff
@@ -28,16 +28,23 @@ type AmazonS3Storage = {
...
 };
 
-type Storage = LocalStorage | SftpStorage | AmazonS3Storage | NoneStorage;
+type Storage = LocalStorage | SftpStorage | AmazonS3Storage | MicrosoftAzureStorage | NoneStorage;
 
-type StorageType = 'none' | 'local' | 'sftp' | 'amazon_s3';
+type StorageType = 'none' | 'local' | 'sftp' | 'amazon_s3' | 'microsoft_azure';
 
-const STORAGE_TYPES = ['none', 'sftp', 'amazon_s3'];
+const STORAGE_TYPES = ['none', 'sftp', 'amazon_s3', 'microsoft_azure'];
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/model.ts:47]: THere is something strange here, it miss something. The microsoft_azure, amazin_s3 and sftp should be only available on EE and not in CE. There is a feature flag before it's strange that there is no more feature flag ! The code will not work on CE (the route to check connection does not exist for exemple)

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.test.tsx [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+    ),
+    'name'
+  );
+
+  expect(onStorageChange).toHaveBeenLastCalledWith({
+    ...storage,
+    container_name: 'container_name',
+  });
+});
+
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.test.tsx:138]: May add a test about the check connection is not available if there is an empty field

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (axelhms) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/front/src/components/StorageConfigurator/MicrosoftAzureStorageConfigurator.tsx:19]: Will be done in [RAB-1236](https://akeneo.atlassian.net/jira/software/projects/RAB/boards/282/backlog?selectedIssue=RAB-1236)
